sql/catalyst/src/main/scala/org/apache/spark/sql/internal/SQLConf.scala

```diff
@@ -2155,6 +2155,15 @@ object SQLConf {
     .booleanConf
     .createWithDefault(true)
 
+  val ADD_PARTITION_BATCH_SIZE =
+    buildConf("spark.sql.addPartitionInBatch.size")
+      .internal()
+      .doc("The number of partitions to be handled in one turn when use " +
+        "`AlterTableAddPartitionCommand` to add partitions into table. The smaller " +
+        "batch size is, the less memory is required for the real handler, e.g. Hive Metastore.")
+      .intConf
```

Comment:
Since this become a variable from now, we need to add `.checkValue(` to prevent accidental user mistake.

Reply:
Maybe, `> 0`? Do you think we have a reasonable upper bound for this, too?

Reply:
Lower bound checking might be enough, done in https://github.com/apache/spark/pull/27413/commits/ac8b771c1264fc8069a61a3cdb338f4ac6c406af.